examples/kv-events/offline/main.go

```diff
@@ -0,0 +1,282 @@
+	}
+
+	return &kvevents.Config{
+		Concurrency: concurrency,
+		ZMQEndpoint: zmqEndpoint,
+		TopicFilter: zmqTopic,
+	}
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting KV Events Pool Example")
+
+	kvCacheIndexer, err := setupKVCacheIndexer(ctx)
+	if err != nil {
+		logger.Error(err, "failed to setup KVCacheIndexer")
+		return
+	}
+
+	// Setup events pool with ZMQ subscriber
+	eventsPool := setupEventsPool(ctx, kvCacheIndexer.KVBlockIndex())
+
+	// Start events pool
+	eventsPool.Start(ctx)
+	logger.Info("Events pool started and listening for ZMQ messages")
+
+	// Setup ZMQ publisher to simulate vLLM engines
+	publisher, err := setupPublisher(ctx)
+	if err != nil {
+		logger.Error(err, "failed to setup ZMQ publisher")
+		return
+	}
+	defer publisher.Close()
+
+	// Setup graceful shutdown
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-sigChan
+		logger.Info("Received shutdown signal")
+		cancel()
+	}()
+
+	// Run the demonstration
+	if err := runEventsDemo(ctx, kvCacheIndexer, publisher); err != nil {
+		logger.Error(err, "failed to run events demo")
+		return
+	}
+
+	// Wait for shutdown signal
+	<-ctx.Done()
+	logger.Info("Shutting down...")
+
+	// Graceful shutdown of events pool
+	shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer shutdownCancel()
+	eventsPool.Shutdown(shutdownCtx)
+}
+
+func setupKVCacheIndexer(ctx context.Context) (*kvcache.Indexer, error) {
+	logger := klog.FromContext(ctx)
+
+	//nolint:contextcheck // NewKVCacheIndexer does not accept context parameter
+	kvCacheIndexer, err := kvcache.NewKVCacheIndexer(getKVCacheIndexerConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info("Created Indexer")
+
+	go kvCacheIndexer.Run(ctx)
+	modelName := getModelName()
+	logger.Info("Started Indexer", "model", modelName)
+
+	return kvCacheIndexer, nil
+}
+
+func setupEventsPool(ctx context.Context, kvBlockIndex kvblock.Index) *kvevents.Pool {
+	logger := klog.FromContext(ctx)
+
+	cfg := getEventsPoolConfig()
+
+	logger.Info("Creating events pool", "config", cfg)
+	pool := kvevents.NewPool(cfg, kvBlockIndex)
+
+	return pool
+}
+
+func setupPublisher(ctx context.Context) (*Publisher, error) {
+	logger := klog.FromContext(ctx)
+
+	cfg := getEventsPoolConfig()
+
+	logger.Info("Creating ZMQ publisher (simulating vLLM engines)", "endpoint", cfg.ZMQEndpoint)
+
+	publisher, err := NewPublisher(cfg.ZMQEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ZMQ publisher: %w", err)
+	}
+
+	logger.Info("ZMQ publisher created successfully")
+	return publisher, nil
+}
+
+func runEventsDemo(ctx context.Context, kvCacheIndexer *kvcache.Indexer, publisher *Publisher) error {
+	logger := klog.FromContext(ctx)
+
+	modelName := getModelName()
+	logger.Info("@@@ Starting KV Events Demo", "model", modelName)
```

Comment:
nit: why the `@@@` style in logging?

Reply:
Made it easier to track, especially if you enable debug/trace logging.